### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by silviatofana**
At this point your codebase is pretty big. The main.rb file may be especially large at this point. Let's try to clean things up!

Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
Follow SOLID principles.
Pay special attention to the "Single-responsiblity Principle".
Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,133 +1,110 @@
...
+  def menu_selection
+    main_menu
+    selection = gets.chomp.to_i
+    case selection
+    when 1
+      @books.list_all_books
+    when 2
+      list_all_people
+    when 3
+      create_person
```
Comment (dasileker) [inline app.rb]: - [ ] Hello @silviatofana Nice job on refactoring but this part should be placed in the `main.rb` file

File: app.rb [hunk 2]
```diff
@@ -1,133 +1,110 @@
...
-
-    person_id = gets.chomp.to_i
-
+  def create_rental()
+    puts 'Select a book from the following list by the number:📖'
+    list_all_books
+    book_number = gets.chomp.to_i
+    puts ' '
+    puts 'Select a person from the following list by the number:🧑‍🎓🧑‍🏫'
+    list_all_people
```
Comment (dasileker) [inline app.rb]: ### **Optional**
- [ ] Please try as much as possible to use `SOLID`  principles to refactor their classes. [SEE THIS ARTICLE](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

File: app.rb [hunk 3]
```diff
@@ -1,133 +1,110 @@
...
+      puts "#{name.capitalize} added successfully 💓"
+    when 2
+      print 'Enter teacher name: '
+      name = gets.chomp
+      print 'Enter age: '
+      age = gets.chomp
+      print 'Enter specialization: '
+      specialization = gets.chomp
+      @people.push(Teacher.new(specialization, age, name))
+      puts "#{name.capitalize} added successfully 💓"
```
Comment (dasileker) [inline app.rb]: ### **Optional**
- [ ] Same goes here try as much as possible.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @silviatofana,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Liters are green✅.
- PR is very Pro✅.
- Commit messages are consistent👌.

## Required Changes ♻️
_Check the comments under the review._

- **Please take a look at the inline comments the ones marked as `Optional` can be very helpful to make your code look better and also you will gain more perspectives on how to make the codes efficient in ruby.** 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @silviatofana, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by silviatofana** [comment]
thanks

**Pull request merged**